### PR TITLE
add User from DB auth

### PR DESCRIPTION
**Pull request by KovalAlexey**

### CODE REVIEW

File: src/main/java/com/dev/cinema/config/AppConfig.java [hunk 1]
```diff
@@ -14,7 +14,9 @@
 @PropertySource("classpath:db.properties")
 @ComponentScan(basePackages = {
         "com.dev.cinema.service",
+        "com.dev.cinema.security",
         "com.dev.cinema.dao"
+
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/config/AppConfig.java]: don't add this empty line, it's redundant

File: src/main/java/com/dev/cinema/config/SecurityConfig.java [hunk 2]
```diff
@@ -2,32 +2,49 @@
...
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
 
+    @Autowired
+    private UserDetailsService userDetailsService;
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/config/SecurityConfig.java]: it's better to use constructor injection

File: src/main/java/com/dev/cinema/controllers/InjectData.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import com.dev.cinema.service.ShoppingCartService;
+import com.dev.cinema.service.UserService;
+import java.util.Set;
+import javax.annotation.PostConstruct;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+@Component
+public class InjectData {
+    private static final String ADMIN = "ADMIN";
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/controllers/InjectData.java]:
```suggestion
    private static final String ADMIN_ROLE_NAME = "ADMIN";
```

File: src/main/java/com/dev/cinema/controllers/InjectData.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    private static final String USER = "USER";
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/controllers/InjectData.java]: same

File: src/main/java/com/dev/cinema/controllers/InjectData.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    private static final Logger log = Logger.getLogger(CinemaHallDaoImpl.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/controllers/InjectData.java]: cineamHall ?

File: src/main/java/com/dev/cinema/controllers/InjectData.java [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    public InjectData(UserService userService,
+                      ShoppingCartService shoppingCartService,
+                      RoleService roleService) {
+        this.userService = userService;
+        this.shoppingCartService = shoppingCartService;
+        this.roleService = roleService;
+    }
+
+    @PostConstruct
+    public void addAdmin() {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/controllers/InjectData.java]: do you add only admin here?

File: src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl implements RoleDao {
+    private static final Logger log = Logger.getLogger(OrderDaoImpl.class);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java]: logger declaration

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.findByEmail(email).get();
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: use orElseThrow with UsernameNotFoundException here

**Review by RomanDubovskyi — APPROVED** [review]
fix comments and you can merge PR without second review

File: src/main/java/com/dev/cinema/config/SecurityConfig.java (hunk 2 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/config/SecurityConfig.java]: done

File: src/main/java/com/dev/cinema/controllers/InjectData.java (hunk 3 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/controllers/InjectData.java]: done

File: src/main/java/com/dev/cinema/controllers/InjectData.java (hunk 4 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/controllers/InjectData.java]: done

File: src/main/java/com/dev/cinema/controllers/InjectData.java (hunk 6 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/controllers/InjectData.java]: done

File: src/main/java/com/dev/cinema/controllers/InjectData.java (hunk 5 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/controllers/InjectData.java]: done

File: src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java (hunk 7 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/dao/impl/RoleDaoImpl.java]: done

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java (hunk 8 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: done

File: src/main/java/com/dev/cinema/config/AppConfig.java (hunk 1 above)
Comment (KovalAlexey) [reply src/main/java/com/dev/cinema/config/AppConfig.java]: done

**Pull request merged**